### PR TITLE
feat(TDI-44483) : Migration test connector

### PR DESCRIPTION
**Pull request by ypiel-talend**
Create a dedicated connector to validation migration in TCK integration (TDC/TPD):
https://jira.talendforge.org/browse/TDI-44483

### CODE REVIEW

File: migration-tester/src/main/java/org/talend/components/migration/conf/DSE.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @Documentation("New  property to duplicate legacy one.")
+    String dse_duplication;
+
+    @Option
+    @Documentation("The legacy property of the dataset.")
+    String dse_legacy;
+
+    @Option
+    @Documentation("Dataset migration handler callback")
+    String dse_migration_handler_callback;
```
Comment (undx) [inline migration-tester/src/main/java/org/talend/components/migration/conf/DSE.java:45]: why have chosen this kind of naming?

File: migration-tester/src/main/java/org/talend/components/migration/migration/AbstractTestMigrationHandler.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.Map;
+
+public abstract class AbstractTestMigrationHandler implements MigrationHandler {
+
+    private static String current;
+
+    static {
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+        current = sdf.format(new Date());
+    }
```
Comment (undx) [inline migration-tester/src/main/java/org/talend/components/migration/migration/AbstractTestMigrationHandler.java:29]: could do it in member

File: migration-tester/src/main/java/org/talend/components/migration/sink/DummySink.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+@Slf4j
+@Version(value = AbstractConfig.VERSION, migrationHandler = SinkMigrationHandler.class)
+@Icon(Icon.IconType.STAR)
+@Processor(name = "DummySink")
+@Documentation("")
+public class DummySink implements Serializable {
+
+    private final AbstractConfig config;
+
+    private transient boolean done = false;
```
Comment (undx) [inline migration-tester/src/main/java/org/talend/components/migration/sink/DummySink.java:38]: is it used anywhere?

File: migration-tester/pom.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>${surefire.version}</version>
+                <configuration>
+                    <systemPropertyVariables>
+                        <talend.component.junit.handler.state>static</talend.component.junit.handler.state>
+                        <talend.junit.beam.version>${beam.version}</talend.junit.beam.version>
```
Comment (undx) [inline migration-tester/pom.xml:33]: is that really needed according the tests?

File: migration-tester/src/main/java/org/talend/components/migration/conf/DSE.java (hunk 1 above)
Comment (ypiel-talend) [reply migration-tester/src/main/java/org/talend/components/migration/conf/DSE.java]: it is all explained in README.md

File: migration-tester/src/main/java/org/talend/components/migration/migration/AbstractTestMigrationHandler.java (hunk 2 above)
Comment (ypiel-talend) [reply migration-tester/src/main/java/org/talend/components/migration/migration/AbstractTestMigrationHandler.java:29]: doesn"t exist anymore

File: migration-tester/src/main/java/org/talend/components/migration/sink/DummySink.java (hunk 3 above)
Comment (ypiel-talend) [reply migration-tester/src/main/java/org/talend/components/migration/sink/DummySink.java:38]: deleted

File: migration-tester/pom.xml (hunk 4 above)
Comment (ypiel-talend) [reply migration-tester/pom.xml:33]: deleted

File: migration-tester/src/main/java/org/talend/components/migration/migration/AbstractTestMigrationHandler.java (hunk 2 above)
Comment (undx) [reply migration-tester/src/main/java/org/talend/components/migration/migration/AbstractTestMigrationHandler.java:29]: still here, no?

File: migration-tester/README.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
+# Migration tester
+
+Here is a PDF that explain TCK migration mechanism : https://talend365.sharepoint.com/:b:/s/component/EUVQqToC-ptJrQFy6rhbm7cBAqkEp-46TeN2-Hle3ZhTAw?e=le5ok4
```
Comment (undx) [inline migration-tester/README.md:3]: This document is not available for talend org non-members and this repo is public.

**Review by undx — DISMISSED** [review]
Minor comments but LGTM

File: migration-tester/README.md (hunk 5 above)
Comment (ypiel-talend) [reply migration-tester/README.md:3]: you're right, better to remove this link.

**Review by undx — APPROVED** [review]
LGTM

**Pull request merged**